frontend/src/components/project_assignment/assignmentItem.tsx

```diff
@@ -1,94 +1,122 @@
+import { deleteAssignment, getAllAssignmentsFormLinks } from "../../api/calls/AssignmentCalls";
 import WarningToast from "../warningToast";
 import useTranslation from "next-translate/useTranslation";
 import { capitalize } from "../../utility/stringUtil";
-import { getSkillTypeFromSkill } from "../../api/calls/skillTypeCalls";
-import { useEffect, useState } from "react";
-import applicationPaths from "../../properties/applicationPaths";
-import { extractIdFromStudentUrl } from "../../api/calls/studentCalls";
+import { IAssignment } from "../../api/entities/AssignmentEntity";
+import AssignmentStudentRow from "./assignmentStudentRow";
+import { useState } from "react";
+import SkillBadge from "../skillBadge";
+
+export interface IAssignmentItemProps {
+    skill: IProjectSkill;
+}
+
+/**
+ * Custom sort function for assignment items.
+ * It sorts the assignments on the students first name,
+ * if the students have the same name we sort on the username of the assigner.
+ * @param assignments list of [IAssignment] that should be sorted in place.
+ * @param keyHolder the keyHolder object that enables the sort on fields not contained within [IAssignment].
+ */
+function sortAssignments(assignments: IAssignment[], keyHolder: IStudentAssignmentSortKeyHolder) {
+    assignments.sort((assign1, assign2) => {
+        const key1 = keyHolder[assign1._links.self.href];
+        const key2 = keyHolder[assign2._links.self.href];
+
+        if (key1 === undefined) {
+            if (key2 === undefined) {
+                return 0;
+            }
+            return -1;
+        } else if (key2 === undefined) {
+            return 1;
+        }
+
+        const compareStudents: number = key1.studentFirstName.localeCompare(key2.studentFirstName);
+        if (compareStudents === 0) {
+            return key1.assignerCallName.localeCompare(key2.assignerCallName);
+        }
+        return compareStudents;
+    });
+}
+
+export interface IAssignmentSortKey {
+    studentFirstName: string;
+    assignerCallName: string;
+}
+
+interface IStudentAssignmentSortKeyHolder {
+    [k: string]: IAssignmentSortKey | undefined;
+}
 
 /**
  * An item containing the information about the assignments to a skill of a project.
  * It contains the name of the skill followed by a list of assignments for that skill.
  * The list contains the first name of the student, the name of the person that did the assignment and the motivation.
  * Furthermore, it contains a delete button to delete the assignment.
- * @param item Skill you want the assignments from
+ * @param skill [IProjectSkill] you want the assignments from
  * @constructor
  */
-function AssignmentItem(item: { skill: IProjectSkill }) {
+function AssignmentItem({ skill }: IAssignmentItemProps) {
     const { t } = useTranslation("common");
+    const { mutate } = useSWRConfig();
+    const { data: resAssignments, error: assignmentsError } = useSWR(
```

Comment:
What does "resAssignments" stand for?

Reply:
recievedAssignment. Do you have a better suggestion?

Reply:
receivedAssignments sounds good to me